server/src/controllers/maintenance.controller.ts

```diff
@@ -1,19 +1,66 @@
-import { BadRequestException, Body, Controller, Post, Res } from '@nestjs/common';
-import { ApiTags } from '@nestjs/swagger';
+import {
+  BadRequestException,
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Post,
+  Res,
+  UploadedFile,
+  UseInterceptors,
+} from '@nestjs/common';
+import { FileInterceptor } from '@nestjs/platform-express';
+import { ApiBody, ApiConsumes, ApiTags } from '@nestjs/swagger';
 import { Response } from 'express';
 import { Endpoint, HistoryBuilder } from 'src/decorators';
 import { AuthDto } from 'src/dtos/auth.dto';
-import { MaintenanceAuthDto, MaintenanceLoginDto, SetMaintenanceModeDto } from 'src/dtos/maintenance.dto';
+import {
+  MaintenanceAuthDto,
+  MaintenanceIntegrityResponseDto,
+  MaintenanceListBackupsResponseDto,
+  MaintenanceLoginDto,
+  MaintenanceStatusResponseDto,
+  MaintenanceUploadBackupDto,
+  SetMaintenanceModeDto,
+} from 'src/dtos/maintenance.dto';
 import { ApiTag, ImmichCookie, MaintenanceAction, Permission } from 'src/enum';
-import { Auth, Authenticated, GetLoginDetails } from 'src/middleware/auth.guard';
+import { Auth, Authenticated, FileResponse, GetLoginDetails } from 'src/middleware/auth.guard';
+import { StorageRepository } from 'src/repositories/storage.repository';
 import { LoginDetails } from 'src/services/auth.service';
 import { MaintenanceService } from 'src/services/maintenance.service';
 import { respondWithCookie } from 'src/utils/response';
+import { FilenameParamDto } from 'src/validation';
 
 @ApiTags(ApiTag.Maintenance)
 @Controller('admin/maintenance')
 export class MaintenanceController {
-  constructor(private service: MaintenanceService) {}
+  constructor(
+    private service: MaintenanceService,
+    private storageRepository: StorageRepository,
+  ) {}
+
+  @Get('status')
+  @Endpoint({
+    summary: 'Get maintenance mode status',
+    description: 'Fetch information about the currently running maintenance action.',
+    history: new HistoryBuilder().added('v9.9.9').alpha('v9.9.9'),
+  })
+  maintenanceStatus(): MaintenanceStatusResponseDto {
+    return {
+      action: MaintenanceAction.End,
+    };
+  }
+
+  @Get('integrity')
+  @Endpoint({
+    summary: 'Get integrity and heuristics',
+    description: 'Collect integrity checks and other heuristics about local data.',
+    history: new HistoryBuilder().added('v9.9.9').alpha('v9.9.9'),
+  })
+  integrityCheck(): Promise<MaintenanceIntegrityResponseDto> {
+    return this.service.integrityCheck();
+  }
```

Comment:
I think that an integrity feature and report has nothing to do with maintenance mode, which is used for running the server in a restricted mode. Can we put this endpoint in `ReportController` at `GET /reports/integrity`, `ReportService.getIntegrityReport()`, etc.?